app/policies/order_policy.rb

```diff
@@ -0,0 +1,28 @@
+class OrderPolicy < ApplicationPolicy
+  class Scope < Scope
+    def resolve
+      scope
+    end
+  end
+
+  def create?
+    create_or_destroy?
+  end
+  
+  def destroy?
+    create_or_destroy?
+  end  
+  
+  def create_or_destroy?
```

Comment:
create_or_destroy?` actually means in this case `create_or_destroy_or_checkout_or_charge`, see what I mean?

avoid method names which are "lies".

Rename it to `touch?`